Apps/WebClient/src/ClientApp/src/components/modal/authenticateRapidTest.vue

```diff
@@ -0,0 +1,398 @@
+            this.$v.$reset();
+        });
+    }
+
+    private isValid(param: Validation): boolean | undefined {
+        return param.$dirty ? !param.$invalid : undefined;
+    }
+
+    private validations() {
+        return {
+            rapidTest: {
+                labSerialNumber: {
+                    required: required,
+                },
+                dateTestTaken: {
+                    required: required,
+                    maxValue: (value: string) =>
+                        new DateWrapper(value).isBefore(new DateWrapper()),
+                },
+                positive: { required: required },
+            },
+        };
+    }
+    private handleSubmit(bvModalEvt: Event) {
+        // Prevent modal from closing
+        bvModalEvt.preventDefault();
+        this.$v.$touch();
+        if (this.$v.$invalid) {
+        } else {
+            this.$v.$reset();
+            this.submitRapidTest();
+        }
+    }
+
+    private submitRapidTest() {
+        this.isLoading = true;
+        SnowPlow.trackEvent({
+            action: "submit",
+            text: "Autheticate_Rapid_Test",
+        });
+        this.laboratoryService
+            .postAutheticateRapidTest(this.patientData.hdid, this.rapidTest)
+            .then((result) => {
+                if (result.resultStatus === ResultType.Success) {
+                    this.isSuccess = true;
+                    this.errorMessage = "";
+                    this.logger.debug(`Rapid Test is submitted." ${result} `);
+                } else {
+                    this.isSuccess = false;
+                    this.errorMessage = `${result.resultError?.resultMessage}`;
+                    this.logger.debug(
+                        `Rapid Test is not submitted." ${result.resultError?.resultMessage} `
+                    );
+                }
+            })
+            .catch((err: ResultError) => {
+                this.isSuccess = false;
+                this.errorMessage = "Error in submitting rapid test";
+                this.logger.error(`Error in submitting rapid test : ${err}`);
+            })
+            .finally(() => {
+                this.isLoading = false;
+            });
+    }
+
+    private isDisabled() {
+        return this.isSuccess;
+    }
+
+    public hideModal(): void {
+        this.$v.$reset();
+        this.isVisible = false;
+    }
+
+    private get showErrorBanner(): boolean {
+        return this.errorMessage !== undefined && this.errorMessage.length > 0;
+    }
+}
+</script>
+
+<template>
+    <b-modal
+        id="covid-test-kit-modal"
+        v-model="isVisible"
+        data-testid="covid-test-kit-modal"
+        content-class="mt-5"
+        title="Submit Rapid Test Result"
+        size="lg"
+        header-bg-variant="primary"
+        header-text-variant="light"
+        centered
+    >
+        <div v-if="isSuccess">
+            <b-alert
+                data-testid="post-Rapid-Test-SuccessBanner"
```

Comment:
nit: naming convention seems to be different from other test ids that use kabab case. Should it be 'post-rapid-test-sucess-banner'?